cpg-library/src/main/java/de/fraunhofer/aisec/cpg/graph/TypeManager.java

```diff
@@ -240,6 +240,10 @@ public ParameterizedType createOrGetTypeParameter(
     return secondOrderTypes;
   }
 
+  public boolean typeExists(String name) {
+    return firstOrderTypes.stream().anyMatch(type -> type.getRoot().getName().equals(name));
```

Comment:
Maybe needs to be synchronised or in a synchronised block, once we merge in #444